src/common/containers/ZeroApprovalModal/ZeroApprovalModal.tsx

```diff
@@ -0,0 +1,46 @@
+import { useWalletDetails, useWalletInfo } from 'modules/wallet'
+import { ConfirmationPendingContent } from 'common/pure/ConfirmationPendingContent'
+import { shortenAddress } from 'legacy/utils'
+import { GpModal } from 'common/pure/Modal'
+import { useWeb3React } from '@web3-react/core'
+import { getStatusIcon } from 'modules/account/containers/AccountDetails'
+import { useZeroApprovalState } from 'common/state/useZeroApprovalState'
+import { useCallback } from 'react'
+
+interface ZeroApprovalModalProps {
+  onDismiss?: () => void
+}
+
+export function ZeroApprovalModal({ onDismiss = () => {} }: ZeroApprovalModalProps) {
+  const { account } = useWalletInfo()
+  const walletDetails = useWalletDetails()
+  const { connector } = useWeb3React()
+  const { isApproving, currency } = useZeroApprovalState()
+
+  const handleDismiss = useCallback(() => {
```

Comment:
The hook looks redundant

Reply:
Not anymore, per a new PR. It was missing a check.